### PR TITLE
Assignment/GroupAssignment slugs should only be unique to the organization

### LINKED ISSUE #342: Project names include some kind of key
Example "myFakeGroup/SuperAwseomeProject-34235-9160-4e50-ba243-545435-cool-team created by mattfxyz

### PR DESCRIPTION
**Pull request by tarebyte**
Closes #342 

~~This still needs tests~~

/cc @johndbritton @mattfxyz

### CODE REVIEW

File: app/models/assignment.rb [hunk 1]
```diff
@@ -41,10 +41,21 @@ def starter_code?
...
+  end
+
   def should_generate_new_friendly_id?
     title_changed?
   end
 
+  def slug_candidates
+    [
+      [:title],
+      [:title, :organization_slug]
```
Comment (johndbritton) [inline app/models/assignment.rb]: Why do we need to include the organization slug here?

Comment (johndbritton) [inline app/models/assignment.rb]: Assignment names shouldn't need to be unique across the app.

**Comment by tarebyte** [comment]
@johndbritton @mattfxyz I fixed the problem, I stripped `friendly_id` out of Classroom and we use our own [slugging method](http://api.rubyonrails.org/classes/ActiveSupport/Inflector.html#method-i-parameterize)

I think this also makes the controllers more clear as to what we are looking for in our parameters now that we explicitly state we are using the slugs.

**Comment by johndbritton** [comment]
:+1: I like this a lot better.

**Comment by tarebyte** [comment]
@johndbritton I added some tests to verify that slugs will be unique when necessary. I think this just needs one more :eyes: and then we can :shipit:

File: spec/models/assignment_spec.rb [hunk 2]
```diff
@@ -3,6 +3,17 @@
 RSpec.describe Assignment, type: :model do
   it_behaves_like 'a default scope where deleted_at is not present'
 
+  describe 'slug uniqueness' do
+    let(:organization) { create(:organization) }
+
+    it 'verifes that the slug is unique even if the titles are unique' do
```
Comment (tarebyte) [inline spec/models/assignment_spec.rb]: @johndbritton this tests seems to satisfy your test case we were discussing.

**Comment by tarebyte** [comment]
@johndbritton this is what the uniqueness check looks like on the form
uniqueness

**Comment by tarebyte** [comment]
This is what title presence validation looks like

presence

**Comment by johndbritton** [comment]
:shipit:

Are you going to follow up with the ID based routing for organizations in another pull request?

**Comment by tarebyte** [comment]
Yup!

**Pull request merged**